Uploard dropdown component

seiyaterada opened the pull request.

BraidenCutforth commented on src/components/drawer-content/DrawerContent.tsx:
```diff
@@ -0,0 +1,199 @@
+)
+
+// CSS Styles
+const drawer_content: CSS.Properties = {
```
We should use camel casing for variables. If your cursor is on the variable and you hit f2 then it will rename it everywhere it is being used.
```suggestion
const drawerContent: CSS.Properties = {
```

cindyunrau commented on src/components/drawer-content/CourseSort.tsx:
```diff
@@ -0,0 +1,6 @@
+import React from 'react'
```
I was going to use this to take the input date range from courses and sort them into their appropriate semesters. Didn't start implementation yet.

BraidenCutforth commented on src/components/drawer-content/DrawerContent.tsx:
```diff
@@ -0,0 +1,179 @@
+    paddingTop: '30px',
+    paddingBottom: '30px',
+    height: '100%',
+}
```
We should use the material-ui styling ways instead of CSS properties. We can convert this pretty easily.  In the useStyles const, add a key like 

```typescript
drawerContent: {
    backgroundColor: '#F5F5F5', // There might be something like theme.pallette.something, if you can find it, great, if not just use this constant for now
    paddingTop: theme.spacing(1), // 1 or more, adjust based on what you want
    paddingBottom: theme.spacing(1), // same as above
    height: '100%',
}
```

Once that is done, then you can use the className property instead of the styles property, suggested below.

BraidenCutforth commented on src/components/drawer-content/DrawerContent.tsx:
```diff
@@ -0,0 +1,179 @@
+    paddingBottom: '30px',
+    height: '100%',
+}
+const textStyles: CSS.Properties = {
```
Similar as above.

BraidenCutforth commented on src/components/drawer-content/DrawerContent.tsx:
```diff
@@ -0,0 +1,179 @@
+    //      On any nested loop click, all nested lists open and close -> should only open and close specific nested list
+
+    return (
+        <div id="drawer_content" style={drawerContent}>
```
Once you change the way the styling is written above, this is the change you need down here
```suggestion
        <div id="drawer_content" className={classes.drawerContent}>
```

daelynj commented on package.json:
```diff
@@ -1,47 +1,47 @@
-    "react": "^16.11.0",
-    "react-dom": "^16.11.0"
-  }
+    "name": "passr",
```
@cindyunrau It looks like your editor applied some formatting to `package.json` and `package-lock.json` by accident. This is avoidable if you only add the files to your commits that you've worked on rather than doing a `git add .` to commit every change.

This has small effects such as changing 7587 lines of code in the repository. Also changing the history of the two files, and when this code gets merged, will change everyone's versions of these files as well. But it's solvable with an interactive rebase if you feel like it!

BraidenCutforth replied:
This was a commit I made to fix some line endings. My editor is set to format json like this on purpose. It probably shouldn't happen here though you're right and we should uncommit this change.

daelynj replied:
@BraidenCutforth My bad, your GitHub profile pictures look pretty much the same, I didn't see you had commits lol

daelynj commented on src/index.css:
```diff
@@ -1,4 +1,5 @@
 /* User Agent Stylesheet Overrides */
 body {
     margin: 0;
+    font-family: 'Roboto arial';
```
I would suggest placing any global changes to CSS like this in a separate PR; while it's a font you want to use, it's not specifically related to the component you are building. It's always best to write small PRs and separate things that are unrelated to each other.

BraidenCutforth replied:
I think this was also me, and again should be separate PR. I think this might actually not be needed though.

daelynj commented on README.md:
```diff
@@ -4,7 +4,7 @@
 In order to develop on this project you need the following:
 
-- [node 12](https://nodejs.org/en/download/)
+-   [node 12](https://nodejs.org/en/download/)
```
Looks like another format thing your editor did by accident 🙃

daelynj commented:
It would be great to have the message body of this PR say what you've created and why you've created it, with a link to a ticket or something of the sort. That would save people a few minutes having to figure it out in the future! A descriptive PR title would help as well, such as "Implement Drawer component" rather than just "Drawer".

seiyaterada committed:
Uploard dropdown component

seiyaterada committed:
Create hooks folder and add ProfileDropdown to molecules

seiyaterada committed:
Merge branch 'master' into PAS-59-Create-Profile-Dropdown

seiyaterada committed:
update profile dropdown

BraidenCutforth committed:
PAS-59: Account Dropdown

- Add types to useComponentVisible hook
- Add ref parameter for useComponentVisible hook
- Simplified state in ProfileDropdown
- Removed state in DropdownList

seiyaterada commented on src/hooks/useComponentVisible.tsx:
```diff
@@ -1,13 +1,15 @@
 
-export default function useComponentVisible(initialIsVisible) {
+export default function useComponentVisible(
+	ref: React.RefObject<HTMLElement>,
```
I'm still not sure what exactly the ref is, and what value it holds in the ref?

BraidenCutforth replied:
@BraidenCutforth

BraidenCutforth replied:
@seiyaterada A ref is esentially a pointer to an HTMLElement. In this case it is a pointer to the container which is using it, to detect clicks outside of it. (Or really detect if the click was inside the container...)

BraidenCutforth replied:
As an added note for this to make sense, the dom is a tree of elements, and an element has a reference to the parent that is above it.

seiyaterada committed:
Create generic profile dropdown

seiyaterada committed:
Merge branch 'master' into PAS-59-Create-Profile-Dropdown

BraidenCutforth committed:
Braiden - Generic Additions

BraidenCutforth committed:
PAS-59: Align the dropdown

seiyaterada committed:
Added icons for Profile dropdown items and minor changes
